v2/spanner-common/src/main/java/com/google/cloud/teleport/v2/spanner/migrations/utils/ShardFileReader.java

```diff
@@ -53,21 +60,124 @@ public ShardFileReader(ISecretManagerAccessor secretManagerAccessor) {
     this.secretManagerAccessor = secretManagerAccessor;
   }
 
-  public List<Shard> getOrderedShardDetails(String sourceShardsFilePath) {
+  public static class ShardConfig {
```

Comment:
I am confused, why did this file need to change? There is nothing that should change with respect to accepting the shard configuration from the user? 
I thought the only thing we were changing was adding a template level flag?